src/algokit-core-bridge/transaction-composer.ts

```diff
@@ -0,0 +1,483 @@
+   * to this account
+   */
+  closeRemainderTo?: Address
+}
+
+export type TransactionWithContext = {
+  txn: Transaction
+  context: {
+    /* The logical max fee for the transaction, if one was supplied. */
+    maxFee?: AlgoAmount
+    // TODO: abiMethod
+    /* The ABI method, if the app call transaction is an ABI method call. */
+    // abiMethod?: algosdk.ABIMethod
+  }
+}
+
+export type TransactionWithSignerAndContext = TransactionWithSigner & TransactionWithContext
+
+export type SendAtomicTransactionComposerResults = {
+  transactions: Transaction[]
+  /** base64 encoded representation of the group ID of the atomic group */
+  groupId: string
+  /** The transaction IDs that have been prepared and/or sent */
+  txIds: string[]
+  /** The responses if the transactions were sent and waited for,
+   * the index of the confirmation will match the index of the underlying transaction
+   */
+  confirmations: algosdk.modelsv2.PendingTransactionResponse[] // TODO: can we convert this yet?
+}
+
+/** Parameters to create an `TransactionComposer`. */
+export type TransactionComposerParams = {
+  /** The algod client to use to get suggestedParams and send the transaction group */
+  algodClient: AlgodClient
+  /** The function used to get the TransactionSigner for a given address */
+  getSigner: (address: string | Address) => TransactionSigner
+  /** The method used to get SuggestedParams for transactions in the group */
+  getTransactionParams?: () => Promise<TransactionParams>
+  /** How many rounds a transaction should be valid for by default; if not specified
+   * then will be 10 rounds (or 1000 rounds if issuing transactions to LocalNet).
+   */
+  defaultValidityWindow?: bigint
+  /**
+   * An array of error transformers to use when an error is caught in simulate or execute
+   * callbacks can later be registered with `registerErrorTransformer`
+   */
+  errorTransformers?: ErrorTransformer[]
+  defaultValidityWindowIsExplicit?: boolean
+}
+
+export type Txn = PaymentTransactionParams & { type: 'pay' }
+
+export class TransactionComposer {
+  private algosdkAlgod: algosdk.Algodv2
+  private algod: AlgodApi
+  private getTransactionParams: () => Promise<TransactionParams>
+  private getSigner: (address: string | Address) => TransactionSigner
+  /** The default transaction validity window */
+  private defaultValidityWindow = 10n
+  /** Whether the validity window was explicitly set on construction */
+  private defaultValidityWindowIsExplicit = false
+  errorTransformers: ErrorTransformer[]
+  private txns: Txn[] = []
+
+  /**
+   * Create a `TransactionComposer`.
+   * @param params The configuration for this composer
+   * @returns The `TransactionComposer` instance
+   */
+  constructor(params: TransactionComposerParams) {
+    this.algod = params.algodClient.algoKitCoreAlgod
+    this.algosdkAlgod = params.algodClient
+
+    const defaultGetTransactionParams = async (): Promise<TransactionParams> => {
```

Comment:
I don't believe this is ever used currently, which means we aren't getting transaction params using the core http client. If we want caching, then doing this could be a little tricky as the cache is stored in the AlgorandClient.